ssh/tailssh/accept_env_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tailssh
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMatchPattern(t *testing.T) {
+	testCases := []struct {
+		pattern string
+		target  string
+		match   bool
+	}{
+		// Positive cases
+		{pattern: "*", target: "EXAMPLE_ENV", match: true},
+		{pattern: "?", target: "A", match: true},
+		{pattern: "?*", target: "EXAMPLE_2", match: true},
+		{pattern: "??", target: "CC", match: true},
+		{pattern: "?*?", target: "ABCDEFG", match: true},
+		{pattern: "*?*", target: "ABCDEFG", match: true},
+		{pattern: "***", target: "123456", match: true},
+		{pattern: "**?TEST", target: "_TEST", match: true},
+		{pattern: "TEST**?", target: "TEST_", match: true},
+		{pattern: "TEST_*", target: "TEST_A", match: true},
```

Comment:
It's technically covered in the below test, but it might be nice to also add a test case where the target is longer than a pattern terminated in `*`.